src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java

```diff
@@ -277,7 +277,9 @@ private void setDependenciesAndArtifacts(Module buildInfoModule, String artifact
 
         List<Dependency> dependencies = new ArrayList<Dependency>();
         List<Artifact> artifacts = new ArrayList<Artifact>();
-        Iterator<String> it = DockerUtils.getLayersDigests(manifest).iterator();
+        // Filter out duplicate layers from manifest by using HashSet.
+        // Docker manifest may hold 'empty layers', as a result, docker promote will fail to promote the same layer more than once.
+        Iterator<String> it = new ArrayList<>(new HashSet<>(DockerUtils.getLayersDigests(manifest))).iterator();
         for (int i = 0; i < dependencyLayerNum; i++) {
```

Comment:
Since you're now using an iterator, you should probably use it to iterate, instead of using the ```dependencyLayerNum``` value.
It should also be safer, because when using the code as is, you're relying on two different iteration methods.

Reply:
This indeed a different iterator but it iterates over dependencies. Look at line 292 which ends the dependencies flow and continues to handle artifacts.
However, looks like the scenario you are talking about could and will result in exception out of bound. I will change this to be 100% bulletproof.